lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -1435,6 +1435,42 @@ ParseResult JoinOp::parse(OpAsmParser &parser, OperationState &result) {
 
 void JoinOp::print(OpAsmPrinter &p) { sost::printOp(p, *this, false); }
 
+/// Based on mlir::func::CallOp::verifySymbolUses
+LogicalResult InstanceOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
+  // Check that the module attribute was specified.
+  auto fnAttr = (*this)->getAttrOfType<FlatSymbolRefAttr>("module");
+  if (!fnAttr)
+    return emitOpError("requires a 'module' symbol reference attribute");
+  FuncOp fn = symbolTable.lookupNearestSymbolFrom<FuncOp>(*this, fnAttr);
+  if (!fn)
+    return emitOpError() << "'" << fnAttr.getValue()
+                         << "' does not reference a valid handshake function";
+
+  // Verify that the operand and result types match the callee.
+  auto fnType = fn.getFunctionType();
+  if (fnType.getNumInputs() != getNumOperands())
+    return emitOpError("incorrect number of operands for module");
```

Comment:
Shouldn't this be `function` (or maybe `handshake function`) rather than `module`?

Reply:
I was referring to the attribute name, but explicitly stating that we are talking about a handshake makes more sense.